File: Java/qaly.java
/*
 * Problem: Quality-Adjusted Life-Year
 */

import java.util.Scanner;

public class qaly {

	public static void main(String[] args) {
		double result = 0;
	
		Scanner sc = new Scanner(System.in);
		double periods = sc.nextDouble();
		
		for (int i = 0; i < periods; i++) {
			double quality = sc.nextDouble();
			double years = sc.nextDouble();
			
			double qualy = quality * years;
			
			result += qualy;
		}
	
		System.out.println(result);
	} // end qaly

}

File: Python/qaly.py
'''
 *
 *	Problem: "Quality-Adjusted Life-Year"
 *	Problem ID: "qaly"
 *	https://open.kattis.com/problems/qaly
 *
 *	Solution: Python
'''

#-----main-----#
result = 0
periods = float( input() )

index = 0
while (index < periods):

	line = input().split()
	quality = float( line[0] )
	years = float( line[1] )
	
	qaly = quality * years
	result += qaly
	
	index += 1
	
print(result)
